fluss-client/src/main/java/com/alibaba/fluss/client/admin/Admin.java

```diff
@@ -244,6 +244,22 @@ CompletableFuture<Void> createTable(
      */
     CompletableFuture<List<PartitionInfo>> listPartitionInfos(TablePath tablePath);
 
+    /**
+     * List all partitions in the given table in fluss cluster asynchronously.
+     *
+     * <p>The following exceptions can be anticipated when calling {@code get()} on returned future.
+     *
+     * <ul>
+     *   <li>{@link TableNotExistException} if the table does not exist.
+     *   <li>{@link TableNotPartitionedException} if the table is not partitioned.
```

Comment:
This interface need to check whether the input partitionSpec is valid, so `{@link InvalidPartitionException} if the input partition spec is invalid.`